### PR TITLE
Fixes #709: Endpoint to update user details

### LINKED ISSUE #709: API Endpoint to change user details
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behavior:**
There is no endpoint to serve updation of user details like password or name.
**Expected behavior:**
There should be an endpoint to serve the purpose.

__Yes__ I am sending PR for the same.

### PR DESCRIPTION
**Pull request by yashLadha**
Fixes #709 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

### Preview Link 

- **Replace XXX with your PR no**
- Link to live demo: http://pr-XXX-fossasia-badgeyay.surge.sh  

#### Changes proposed in this pull request:

- Endpoint to update the user details.
-

### CODE REVIEW

**Comment by yashLadha** [comment]
Please review @vaibhavsingh97 @ParthS007 @gabru-md @djmgit

**Comment by yashLadha** [comment]
Please see video for the review. :smile: 
https://www.youtube.com/watch?v=w37p59jhsOs&feature=youtu.be

File: backend/blueprint/api/controllers/modifyUser.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    if data and data['username']:
+        user = User.getUser(data['username'])
+        if user:
+            if not verifyPassword(user, data['password']):
+                return jsonify(Response(401).generateMessage('Wrong username and password combination'))
+            user.name = data['name']
+            try:
+                user.save_to_db()
+            except Exception as e:
+                print(e)
```
Comment (gabru-md) [inline backend/blueprint/api/controllers/modifyUser.py]: A better way would be to pass the exception as `str(e)` inside the Response(401).generateMessage?
What are your thoughts?

File: backend/blueprint/api/controllers/modifyUser.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+
+@router.route('/name', methods=['PUT'])
+def changeName():
+    data = request.get_json()
+    if data and data['username']:
+        user = User.getUser(data['username'])
+        if user:
+            if not verifyPassword(user, data['password']):
+                return jsonify(Response(401).generateMessage('Wrong username and password combination'))
```
Comment (gabru-md) [inline backend/blueprint/api/controllers/modifyUser.py]: Can there be proper indentation just like other files?
Please see this : [https://github.com/fossasia/badgeyay/blob/development/backend/blueprint/api/controllers/generateBadges.py#L9](https://github.com/fossasia/badgeyay/blob/development/backend/blueprint/api/controllers/generateBadges.py#L9)

Comment (gabru-md) [reply backend/blueprint/api/controllers/modifyUser.py]: maybe follow the same approach for every Response sent? :sweat_smile:

File: backend/blueprint/api/controllers/modifyUser.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            user.password = generate_password_hash(data['newPassword'])
+            try:
+                user.save_to_db()
+            except Exception as e:
+                print(e)
```
Comment (gabru-md) [inline backend/blueprint/api/controllers/modifyUser.py]: A better way would be to pass the exception as `str(e)` inside the `Response(401).generateMessage`?
What are your thoughts?

Comment (yashLadha) [reply backend/blueprint/api/controllers/modifyUser.py]: Done :+1:

File: backend/blueprint/api/controllers/modifyUser.py (hunk 1 above)
Comment (yashLadha) [reply backend/blueprint/api/controllers/modifyUser.py]: Done :+1:

**Comment by yashLadha** [comment]
@djmgit Sometimes a session may be opened on someone machine and if there is only option to type new password them that person can easily change the password. If there is a filed to type the old password then there is a double security check as you said. Also in digital payment services like Paytm and many shopping portals they always ask us for old password, if the update from the accounts but not if we update from forgot password. Maybe we can use your approach in the forgot password endpoint. @djmgit

**Comment by djmgit** [comment]
@yashLadha I was also thinking of the same reason, just wanted to confirm :+1:

**Comment by djmgit** [comment]
@yashLadha please rebase your branch.

**Comment by yashLadha** [comment]
Rebase brnach @djmgit GTG . Please merge it.

**Comment by djmgit** [comment]
@yashLadha travis is failing.

**Pull request merged**